File: Ej 37 (Anillos del poder)/Python/Anillos.py
## DeathbatO

# Definir las fuerzas de cada raza
fuerzas = {
    'Pelosos': 1,
    'Sureños buenos': 2,
    'Enanos': 3,
    'Númenóreanos': 4,
    'Elfos': 5,
    'Sureños malos': 2,
    'Orcos': 2,
    'Goblins': 2,
    'Huargos': 3,
    'Trolls': 5
}

# Función para calcular el resultado de la batalla
def resultado_batalla(ejercito_bien, ejercito_mal):
    fuerza_total_bien = sum(fuerzas[raza] for raza in ejercito_bien)
    fuerza_total_mal = sum(fuerzas[raza] for raza in ejercito_mal)
    
    integrantes_bien = len(ejercito_bien)
    integrantes_mal = len(ejercito_mal)
    
    if fuerza_total_bien * integrantes_bien > fuerza_total_mal * integrantes_mal:
        return "El bien triunfa en la batalla!"
    elif fuerza_total_bien * integrantes_bien < fuerza_total_mal * integrantes_mal:
        return "El mal prevalece en la batalla!"
    else:
        return "¡Es un empate!"

# Ejemplo de uso
ejercito_bien = ['Elfos', 'Númenóreanos', 'Pelosos']
ejercito_mal = ['Trolls', 'Orcos', 'Goblins']

print(resultado_batalla(ejercito_bien, ejercito_mal))

File: Ej 37 (Anillos del poder)/Java/Anillos.java
import java.util.HashMap;
import java.util.Map;

public class Anillos{

    // Definir las fuerzas de cada raza
    private static final Map<String, Integer> fuerzas = new HashMap<>();
    static {
        fuerzas.put("Pelosos", 1);
        fuerzas.put("Sureños buenos", 2);
        fuerzas.put("Enanos", 3);
        fuerzas.put("Númenóreanos", 4);
        fuerzas.put("Elfos", 5);
        fuerzas.put("Sureños malos", 2);
        fuerzas.put("Orcos", 2);
        fuerzas.put("Goblins", 2);
        fuerzas.put("Huargos", 3);
        fuerzas.put("Trolls", 5);
    }

    // Función para calcular el resultado de la batalla
    public static String resultadoBatalla(String[] ejercitoBien, String[] ejercitoMal) {
        int fuerzaTotalBien = calcularFuerzaTotal(ejercitoBien);
        int fuerzaTotalMal = calcularFuerzaTotal(ejercitoMal);

        int integrantesBien = ejercitoBien.length;
        int integrantesMal = ejercitoMal.length;

        if (fuerzaTotalBien * integrantesBien > fuerzaTotalMal * integrantesMal) {
            return "El bien triunfa en la batalla!";
        } else if (fuerzaTotalBien * integrantesBien < fuerzaTotalMal * integrantesMal) {
            return "El mal prevalece en la batalla!";
        } else {
            return "¡Es un empate!";
        }
    }

    // Función auxiliar para calcular la fuerza total de un ejército
    private static int calcularFuerzaTotal(String[] ejercito) {
        int fuerzaTotal = 0;
        for (String raza : ejercito) {
            fuerzaTotal += fuerzas.get(raza);
        }
        return fuerzaTotal;
    }

    // Método main para probar el programa
    public static void main(String[] args) {
        String[] ejercitoBien = {"Elfos", "Númenóreanos", "Pelosos"};
        String[] ejercitoMal = {"Trolls", "Orcos", "Goblins"};

        System.out.println(resultadoBatalla(ejercitoBien, ejercitoMal));
    }
}
